nixos/tests/gnome-extensions.nix

```diff
@@ -0,0 +1,153 @@
+import ./make-test-python.nix (
+{ pkgs, lib, ...}:
+{
+  name = "gnome-extensions";
+  meta.maintainers = [ lib.maintainers.piegames ];
+
+  nodes.machine =
+    { pkgs, ... }:
+    {
+      imports = [ ./common/user-account.nix ];
+
+      # Install all extensions
+      environment.systemPackages = lib.pipe pkgs.gnomeExtensions [
+        (lib.filterAttrs (name: _: name != "buildShellExtension")) /* sigh */
+        (lib.filterAttrs (name: _: name != "recurseForDerivations")) /* more sigh */
+        lib.attrValues
+      ];
```

Comment:
What about just `(lib.filterAttrs (name: val: val ? extensionUuid) pkgs.gnomeExtensions)`

Reply:
That's equally hacky to me, but given we're in an untyped language everything's going to be ugly anyways, so I don't mind changing that